### PR TITLE
Aruha-2738 Fix cache nodes deletion behavior

### PR DESCRIPTION
**Pull request by antban**
Aruha-2738 Fix cache nodes deletion behavior

### CODE REVIEW

File: core-services/src/main/java/org/zalando/nakadi/cache/ChangeSet.java [hunk 1]
```diff
@@ -60,9 +61,9 @@ private Date getActualChangeDate(final Change change) {
         for (final Map.Entry<String, List<Change>> newEntry : newChangeMap.entrySet()) {
             // figure out proper dates of the changes
             final List<Change> newChangesSorted = newEntry.getValue().stream()
-                    .sorted(Comparator.comparing(this::getActualChangeDate).reversed())
+                    .sorted(Comparator.comparing(this::getActualChangeDate).thenComparing(Change::getId).reversed())
                     .collect(Collectors.toList());
-            final long newestAge = System.currentTimeMillis() - newChangesSorted.get(0).getOccurredAt().getTime();
+            final long newestAge = System.currentTimeMillis() - getActualChangeDate(newChangesSorted.get(0)).getTime();
```
Comment (antban) [inline core-services/src/main/java/org/zalando/nakadi/cache/ChangeSet.java]: The fix is : 
`newChangesSorted.get(0).getOccurredAt()` -> `getActualChangeDate(newChangesSorted.get(0))`

Comment (antban) [reply core-services/src/main/java/org/zalando/nakadi/cache/ChangeSet.java]: Everything else is to guarantee the same behavior on all the nodes

**Comment by antban** [comment]
:+1:

**Comment by thomasabraham** [comment]
👍

**Pull request merged**